scripts/generate_linux_tests.rb

```diff
@@ -109,13 +116,26 @@ def createLinuxMain(testsDirectory, allTestSubDirectories, files)
     end
 
     for testCase in testCases.sort { |x, y| x <=> y }
-      file.write '    testCase(' + testCase + ".allTests),\n"
+      needsIfGuard = testCaseNeedsIfGuard(testCase, allTestSubDirectories, 5, 1)
+      unless needsIfGuard
+        file.write '    testCase(' + testCase + ".allTests),\n"
+      end
     end
     file.write "])\n"
     file.write "#endif\n"
   end
 end
 
+def testCaseNeedsIfGuard(testCase, allTestSubDirectories, major, minor)
+  for f in allTestSubDirectories
+    if File.exist? "Tests/" + f + "/" + testCase.split(/./, 1).first + ".swift" then
+      return f =~ /#{major}#{minor}/
+    end
+  end
+
+  false
+end
```

Comment:
This is horrible, though good enough to have the discussion if we want to make this workaround less horrible, remove those few tests entirely (and remove this hack), or what else.

Reply:
what is this working around?

Reply:
A bug in Swift 5.0 that it cannot parse/lex code that has the shape of "Thing.$thing", which task-locals have to use - even when surrounded with `#if swift(>=5.1)` swift 5.0 will still fail to compile such code (yeah, it should ignore code in that #if, but it doesn't).

Options have are:
- don't have any tests which show how a task-local can be used with providers (we could not have those tests... and just static providers everywhere in our tests, though it's a bit annoying)
  - (no actual code is 5.0 breaking actually in this PR, it is only the tests)
- or use this separate module workaround

We can still have testing of this pattern in the tracing module, but it's a bit weird to test metadata provider patterns in a different package than the one declaring them.